### PR TITLE
[FIX] account: consider subsequent tax and tags only when next tax is_base_affected is True

### PR DESCRIPTION
**Pull request by jva-odoo**
we have the same functionality before this commit https://github.com/odoo/odoo/commit/ab0bdf0192120671010634978f202da35ddfc79f

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/account/models/account_tax.py [hunk 1]
```diff
@@ -972,6 +972,8 @@ def _prepare_taxes_computation(
             if batch['include_base_amount']:
                 for next_batch in ascending_batches[i + 1:]:
                     for next_tax_data in next_batch['taxes']:
+                        if not next_tax_data['is_base_affected']:
+                            continue
```
Comment (smetl) [inline addons/account/models/account_tax.py:976]: You must reflect the same change in account_tax.js
https://github.com/odoo/odoo/blob/4e10312b14b66df6d9910a714298f5a72d159155/addons/account/static/src/helpers/account_tax.js#L300

File: addons/account/tests/test_invoice_taxes.py [hunk 2]
```diff
@@ -734,3 +734,99 @@ def test_fixed_tax_with_zero_price(self):
...
+        self.assertRecordValues(invoice.line_ids.filtered('tax_line_id'), [
+            {
+                'tax_ids': is_base_affected_true_tax.ids,
+                'tax_tag_ids': self.tax_tag_pos.ids + self.base_tag_pos.ids,
+            },
+            {
+                'tax_ids': [],
+                'tax_tag_ids': self.tax_tag_pos.ids,
+            },
+        ])
```
Comment (smetl) [inline addons/account/tests/test_invoice_taxes.py:832]: Check both in the same method to not duplicate the creation of taxes

**Comment by jva-odoo** [comment]
@smetl it's done

**Review by smetl — APPROVED** [review]
@robodoo r+

**Comment by smetl** [comment]
@jva-odoo Hmmm it seems you have to rebase :(

**Comment by jva-odoo** [comment]
@smetl rebased

**Review by smetl — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**